packages/dds/cell/src/cell.ts

```diff
@@ -242,10 +242,11 @@ export class SharedCell<T extends Serializable = any> extends SharedObject<IShar
      * {@inheritDoc @fluidframework/shared-object-base#SharedObject.loadCore}
      */
     protected async loadCore(storage: IChannelStorageService): Promise<void> {
-        const rawContent = await storage.read(snapshotFileName);
+        const blob = await storage.readBlob(snapshotFileName);
```

Comment:
I really dislike existing duplication of code.
In my view, readBlob, bufferToString & JSON.parse should be refactored to be replaced by calling readAndParse().
If it was done in the first place (before you :)), it would make your work trivial - just change 2 lines of code and all of repo is using different flow.
Same arguments works for future - maybe we will decide that for all or some scenarios we want to encode/decode everything in utf-16 - it would be so much simpler if such logic exists in a single place.

Reply:
so I should change readAndParse to consume readBlob and then replace code like above with it?

Reply:
Yes, that would be preference. It can go in either order - you could first change all this code to use readAndParse, and then change it in later PR.

Reply:
You may get this PR in as is (given it's ready), and following up here with separate PR.